outlook_util.py

```diff
@@ -0,0 +1,101 @@
+import sys
+import json
+import logging
+import datetime
+import requests
+import msal
+import atexit
+import os
+
+
+logging.basicConfig(level=logging.INFO)
+logging.getLogger("msal").setLevel(logging.INFO)  # Optionally disable MSAL DEBUG logs
+
+def get_access_token():
+    mscache = msal.SerializableTokenCache()
+    if os.path.exists("outlooktoken.bin"):
+        mscache.deserialize(open("outlooktoken.bin", "r").read())
+
+    atexit.register(lambda:
+        open("outlooktoken.bin", "w").write(mscache.serialize())
+        if mscache.has_state_changed else None
+        )
+
+    app = msal.PublicClientApplication("3b49f0d7-201a-4b5d-b2b4-8f4c3e6c8a30", authority="https://login.microsoftonline.com/consumers", token_cache=mscache)
```

Comment:
I've set up an Azure application already so the user won't have to.  I'm not worried about rate limits, because it's based on user per client ID, so it's OK.